tests/methods.js

```diff
@@ -0,0 +1,88 @@
+describe('Implementing methods and replies', function () {
+    var connectedBus = null;
+
+    beforeEach(function (done) {
+        AllJoyn.connect(function (bus) {
+            connectedBus = bus;
+            var applicationObjects = [
+                {
+                    path: '/path',
+                    interfaces: [
+                        [
+                            'com.example.methods.interface',
+                            '?DoSomething <s >s',
+                            null
+                        ],
+                        null
+                    ]
+                },
+                null
+            ];
+            var proxyObjects = [
+                {
+                    path: '/path',
+                    interfaces: [
+                        [
+                            'com.example.methods.interface',
+                            '?DoSomething <s >s',
+                            null
+                        ],
+                        null
+                    ]
+                },
+                null
+            ];
+            AllJoyn.registerObjects(function () {
+                done();
+            }, function () {
+                expect(true).toBe(false);
+                done();
+            }, applicationObjects, proxyObjects);
+        });
+    });
+
+    it('should be able to respond to implemented methods', function (done) {
+        var TEST_SERVICE_NAME = 'com.example.application.service';
+        var TEST_SERVICE_PORT = 27;
+
+        var doSomethingHandler = function (messageForReply) {
+            var receivedMessage = messageForReply.message;
+            if (receivedMessage.arguments[0] === 'Incoming success') {
+                messageForReply.replySuccess('s', ['Outgoing success']);
+            } else {
+                messageForReply.replyError('Outgoing error');
+            }
+        };
+        connectedBus.addListenerForReply([1, 0, 0, 0], 's', doSomethingHandler);
+
+        connectedBus.startAdvertisingName(function () {
+            connectedBus.joinSession(function (session) {
+                var callDoSomething = function (incomingString, successCallback, errorCallback) {
+                    session.callMethod(function (message) {
+                        successCallback(message.arguments[0]);
+                    }, function (status) {
+                        errorCallback(status);
+                    }, session.sessionHost,
+                    // TODO: Passing '/path' as parameter instead of null results
```

Comment:
@obsoleted Do you happen to know why I wasn't able to pass '/path' to the `callMethod`? I can try to provide more detailed error trace if that helps.